### PR TITLE
feat: make use-modal non-singleton

### PR DESCRIPTION
**Pull request by vkbansal**
You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- StyleLint: `retrigger stylelint`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- ImageSnapshot `retrigger ImageSnapshot`

### CODE REVIEW

File: packages/useModal/src/useModalHook.tsx [hunk 1]
```diff
@@ -43,30 +35,40 @@ const isFunctionalComponent = (Component: React.FunctionComponent) => {
...
+  const rootRef = useRef<HTMLDivElement | null>()
 
-  useEffect(() => {
-    if (isShown) {
-      context.showModal(key, modal)
-    } else {
-      context.hideModal(key)
+  const showModal = useCallback(() => {
+    const el = document.createElement('div')
+    el.setAttribute('data-id', 'harness-modal-root')
```
Comment (abhinavrastogi-harness) [inline packages/useModal/src/useModalHook.tsx:53]: if there are multiple modals open, data-id will not be unique.. should we add some unique key here?

**Review by abhinavrastogi-harness — DISMISSED** [review]

File: packages/useModal/src/useModalHook.tsx (hunk 1 above)
Comment (vkbansal) [reply packages/useModal/src/useModalHook.tsx]: done

**Review by abhinavrastogi-harness — DISMISSED** [review]

File: packages/useModal/src/__tests__/modal.test.tsx [hunk 2]
```diff
@@ -45,16 +23,17 @@ describe('simple usage', () => {
     return <button onClick={showModal}>Show modal</button>
   }
 
-  test('should show the modal', () => {
-    const { getByText, queryByText } = renderWithProvider(<App />)
+  test.only('should show the modal', async () => {
```
Comment (knagurski) [inline packages/useModal/src/__tests__/modal.test.tsx:26]: Whoops, looks like `only` got through

File: packages/useModal/src/__tests__/modal.test.tsx [hunk 3]
```diff
@@ -78,7 +57,7 @@ describe('simple usage', () => {
   })
 })
 
-describe('updating modal', () => {
+describe.skip('updating modal', () => {
```
Comment (knagurski) [inline packages/useModal/src/__tests__/modal.test.tsx:60]: Do we need this test?

File: packages/useModal/src/__tests__/modal.test.tsx [hunk 4]
```diff
@@ -172,7 +151,7 @@ describe('updating modal', () => {
   })
 })
 
-describe('multiple modals', () => {
+describe.skip('multiple modals', () => {
```
Comment (knagurski) [inline packages/useModal/src/__tests__/modal.test.tsx:154]: Here too

File: packages/useModal/src/useModalHook.tsx [hunk 5]
```diff
@@ -5,8 +5,9 @@
  * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
  */
 
-import { useContext, useEffect, useState, useCallback, useMemo } from 'react'
-import { ModalContext, ModalType } from './ModalContext'
+import React, { useCallback, useEffect, useMemo, useRef } from 'react'
+import { unmountComponentAtNode } from 'react-dom'
+import ReactDOM from 'react-dom'
```
Comment (knagurski) [inline packages/useModal/src/useModalHook.tsx:10]: You can combine there two
```suggestion
import ReactDOM, { unmountComponentAtNode } from 'react-dom'
```

File: src/components/StepWizard/example/StepWizardExample.tsx [hunk 6]
```diff
@@ -82,7 +82,7 @@ export const ExampleWizard = (): JSX.Element => {
   )
 }
 
-const ExampleModal = () => {
+export const ModalExample = (): JSX.Element => {
```
Comment (knagurski) [inline src/components/StepWizard/example/StepWizardExample.tsx:85]: Just curious why we use `JSX.Element` instead of `ReactElement` or `ReactNode`?

**Review by knagurski — COMMENTED** [review]
Looks like some `only` and `skip`s got missed in the tests. Also a question about the `useCallback` reps, but apart from that, a nice simplification and a big improvement 👍🏻

**Pull request closed without merging**